lib/Target/Sophon/BM188x/BM188xEncodeInstsPass.cpp

```diff
@@ -0,0 +1,60 @@
+//===---------------------------------------------------------------------===//
+//
+//                             The ONNC Project
+//
+// Copyright(c) 2018, The ONNC Team
+//
+// This file is part of the ONNC Project and is distributed under
+// 3-clause BSD license (https://opensource.org/licenses/BSD-3-Clause)
+//
+// See LICENSE.TXT for details.
+//
+//===---------------------------------------------------------------------===//
+
+#include "BM188xEncodeInstsPass.h"
+#include <onnc/IR/Compute/Initializer.h>
+#include <onnc/IR/Compute/InputOperator.h>
+#include <onnc/IR/Compute/OutputOperator.h>
+#include <onnc/Target/Sophon/BM188x/bmkernel_api.h>
+
+using namespace onnc;
+using namespace onnc::BM188X;
+
+char BM188xEncodeInsts::ID = 0;
+
+BM188xEncodeInsts::BM188xEncodeInsts(BM188X::CodeEmitVisitor *pInstVisitor,
+                                     const std::string &pFilename)
+    : EncodeInstructions(pInstVisitor, ID), m_FileName(pFilename)
+{
+}
+
+Pass::ReturnType BM188xEncodeInsts::runOnModule(::onnc::Module &pModule)
+{
+  OFStream ofs;
+  std::ostream *os = &onnc::outs();
```

Comment:
It's confusing. You can work on OFStream level directly. That is, pass FILENO_STDOUT to ofs when the file name is "-".

BTW, we prefer use positive condition rather than negative condition. Because in English, we will say if the file name is "-", then we use standard output. We won't use double negatives. We won't say if the file name is NOT "-", then we DON'T use standard output.